idaes/apps/flexibility_analysis/check_optimal.py

```diff
@@ -0,0 +1,32 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+This module provides a function like Pyomo's assert_optimal_termination but that 
+works for both APPSI solver interfaces and non-appsi solver interfaces.
+"""
+import pyomo.environ as pe
+from pyomo.contrib import appsi
```

Comment:
A minor comment (so does not need to be fixed), but this seems like it would be better located in Pyomo (probably in `pyomo.contrib` with the `appsi` code seeing as it related to that).

Reply:
Great point.